anvil/src/eth/error.rs

```diff
@@ -0,0 +1,173 @@
+//! Aggregated error type for this module
```

Comment:
we should make sure these errors are matching geth's errors? is it worth pushing any of them to the RPC crate?